superset/models/core.py

```diff
@@ -362,6 +362,18 @@ def get_effective_user(self, object_url: URL) -> Optional[str]:
             else None
         )
 
+    @contextmanager
+    def get_sqla_engine_with_context(
```

Comment:
What is the difference between having this new `get_sqla_engine_with_context` with the decorator VS using it in our existing `get_sqla_engine` ? I Mean, couldn't we just use the existing one and make use or not of the new functionality the decorator brings when needed? or would that mean changing a ton of places where the `get_sqla_engine` is being used right now?

Reply:
For now there's no difference, but Hugh is planning to add support for SSH tunneling, which would require a setup phase before the engine is created, and a teardown after. In order for the SSH tunnel to work everywhere we will need to replace all existing calls with the new context manager.

Reply:
Oh ok ok, makes sense then 😎 thanks!